PickMeUp/PickMeUp/PMConversationViewController.m

```diff
@@ -5,33 +5,39 @@
 //  Created by Abigail Shilts on 7/14/22.
 //
 
-#import "ConversationViewController.h"
+#import "PMConversationViewController.h"
+#import "PMDirectMessage.h"
 #import "PMDMCell.h"
-#import "DirectMessage.h"
 #import "StringsList.h"
 
-@interface ConversationViewController () <UITableViewDataSource, UITableViewDelegate, UITextViewDelegate>
+@interface PMConversationViewController () <UITableViewDataSource, UITableViewDelegate, UITextViewDelegate>
 @property (strong, nonatomic) IBOutlet UITableView *tableView;
 @property (strong, nonatomic) IBOutlet UILabel *recieverName;
 @property (strong, nonatomic) IBOutlet UITextView *messageToSend;
-@property (strong, nonatomic) NSMutableArray<DirectMessage *> *arrayOfDMs;
+@property (strong, nonatomic) IBOutlet UIButton *mybtn;
+@property (strong, nonatomic) NSMutableArray<PMDirectMessage *> *arrayOfDMs;
 @property BOOL noDMs;
 @end
 
-@implementation ConversationViewController
+@implementation PMConversationViewController
 
-static const NSString *const convoId = @"convoId";
-static const NSString *const cellIdentifier = @"DMCell";
-static const NSString *const postingErr = @"error on Post request";
+static const NSString *const kConvoIdKey = @"convoId";
+static const NSString *const kCellIdentifier = @"DMCell";
+static const NSString *const kPostingErrString = @"error on Post request";
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    self.recieverName.text = self.reciever.username;
+    self.recieverName.text = self.receiver.username;
     self.tableView.dataSource = self;
     self.tableView.delegate = self;
     self.tableView.rowHeight = UITableViewAutomaticDimension;
     self.messageToSend.delegate = self;
-    [self getQuery];
+    [NSTimer scheduledTimerWithTimeInterval:5 target:self selector:@selector(onTimer) userInfo:nil repeats:true];
```

Comment:
Are you polling the server while user is on the conversation screen? Move 5 to a constant like kConversationPollIntervalInSeconds

Reply:
I'd recommend to same NSTimer instance and stop the timer when it's not needed. But maybe you always want it running, not sure.